### ISSUE TITLE
I cannot make heads or tails of how to use gatsby-image-sharp

### ISSUE THREAD

**Issue by doctorpangloss** [gatsbyjs/gatsby#16537]
It's very obscure.

My expectation is that I can specify things very similar to `background-` commands, which is a really useful and familiar way to name image transformation tasks.

Instead, I'm making GraphQL queries? On images? And the side effect of this is that they are processed? And the argument to the React component is `fixed=` or `fluid=` ...?

It's all very mired in its own jargon.

I'm having a great time with the library, but I wonder why you can't do...

```
import Img from "gatsby-image"
import preProcessedImageSrc from "../images/sourceImage.jpg"

...
<Img src={preProcessedImageSrc} fixed maxWidth={100} />
```

...and the library

 1. Does all the post-processing.
 2. Does whatever `graphql` stuff it wants, I don't care.
 3. Builds this `graphql` query based on the argument I give to `src` or just looks it up in a plain old object.
 4. Does what it needs to do.

`graphql` is just so profoundly obscure, it's the XSLT of our time. Leaning on it for image processing tasks is the one place I feel I'm being let down.

**Comment by matowang** [comment]
This is because the lower resolution images for lazy loading have to be created at compile time. Imagine changing the Img src attribute at a state update. It just wouldn't work.

**Comment by freiksenet** [comment]
Gatsby processes images on compile time, so it needs to know at that point what images to create. That's why GraphQL is helpful, because we can execute it on compile time. There are no conditionals inside GraphQL and GraphQL queries are on top page level. This means there can't be any branching that we won't reach during SSR. Code your suggested would fail if it's not in an "initial render". So any conditionals inside React code can break it. 

So to summarize - we use GraphQL because it lets us know what things to generate during compile time, as GraphQL is avaliable on top level and is unambiguous, unlike normal code.

Hope that helps!

**Comment by doctorpangloss** [comment]
Ah, I see. My mistake was thinking it was XSLT, it's more like the Windows Registry of the web! 😄 

I get it, thanks for the kind explanation. I look forward to being a helpful contributor and appreciate all the work that you've done, I figured out how to use the tooling and it all works great.
